packages/ethers-provider/src/provider.ts

```diff
@@ -0,0 +1,45 @@
+import { providers } from 'ethers';
+
+import { state } from './store';
+import type { Chain, ChainId } from './types';
+
+export const ethersProviders: {
+  [key: string]: providers.StaticJsonRpcProvider;
+} = {};
+
+export function getProvider(): providers.StaticJsonRpcProvider;
+export function getProvider(
+  chain: ChainId | Chain,
+): providers.StaticJsonRpcProvider;
+
+export function getProvider(
+  chainOrChainId?: ChainId | Chain,
+): providers.StaticJsonRpcProvider {
+  let chain: Chain;
+
+  if (chainOrChainId === undefined) {
+    chain = state.get().chains[0] as Chain;
+  }
+
+  if (typeof chainOrChainId === 'string') {
+    chain = state.get().chains.find(c => c.id === chainOrChainId) as Chain;
+  }
+
+  if (typeof chainOrChainId === 'object') {
+    chain = chainOrChainId as Chain;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  if (!chain) return null;
+
+  if (!ethersProviders[chain.rpcUrl]) {
+    ethersProviders[chain.rpcUrl] = new providers.StaticJsonRpcProvider(
+      chain.providerConnectionInfo && chain.providerConnectionInfo.url
+        ? chain.providerConnectionInfo
+        : chain.rpcUrl,
+    );
+  }
+
+  return ethersProviders[chain.rpcUrl];
+}
```

Comment:
```suggestion
export function getProvider(
  chainOrChainId?: ChainId | Chain,
): providers.StaticJsonRpcProvider | null {
  let chain: Chain = null;

  if (chainOrChainId === undefined) {
    chain = state.get().chains[0] as Chain;
  }

  if (typeof chainOrChainId === 'string') {
    chain = state.get().chains.find(c => c.id === chainOrChainId) as Chain;
  }

  if (typeof chainOrChainId === 'object') {
    chain = chainOrChainId as Chain;
  }

  if (!chain) {
     return null;
  }

  if (!ethersProviders[chain.rpcUrl]) {
    ethersProviders[chain.rpcUrl] = new providers.StaticJsonRpcProvider(
      chain.providerConnectionInfo && chain.providerConnectionInfo.url
        ? chain.providerConnectionInfo
        : chain.rpcUrl,
    );
  }

  return ethersProviders[chain.rpcUrl];
}
```

What do you think about it? Or do you have some reason for the ignore comments?

Reply:
Would like to prevent using `?.` checks everytime provider method is called, as it will return null only when there is no config added by dev.
Maybe `return null as unknown as providers.StaticJsonRpcProvider` would work?

Reply:
I don't like the current solution but let's wait for @soulBit to see what he thinks. I'd rather prefer the previous one but I think that if a method can return `null`, it should be clear from its signature and not by looking at the implementation.

Reply:
Then maybe just throw error, because that case isn't supposed to happen (at least while users themselves can't add their own chains and rpcs)

Reply:
That sounds pretty good.

Reply:
victor's new solution with explicit undefined value and checks for this seems ok 👌🏻